PSKoans/Koans/dbatools/AboutDbaDatabase.ps1

```diff
@@ -0,0 +1,63 @@
+#requires -modules dbatools
+using module PSKoans
+[Koan(Position = 1001)]
+param()
+<#
+    Get-DbaDatabase
+
+    The Get-DbaDatabase command gets SQL database information for each database
+    that is present on the target instance(s) of SQL Server by default. If the 
+    name of the database is provided, the command will return only the specific 
+    database.
+#>
+Describe 'Get-DbaDatabase' {
+
+    #region Mocks
+    Mock -CommandName Get-DbaDatabase -MockWith {
+        Import-Clixml -Path .\PSKoans\Koans\dbatools\Mocks\Database_All.xml
+    } -ParameterFilter { $_.SqlInstance -eq 'localhost' }
+    
+    Mock -CommandName Get-DbaDatabase -MockWith {
+        Import-Clixml -Path .\PSKoans\Koans\dbatools\Mocks\Database_TestDb.xml
+    } -ParameterFilter { $_.SqlInstance -eq 'localhost' -and $_.Database -eq 'testdb' }
+
+    Mock -CommandName Get-DbaDatabase -MockWith {
+        Import-Clixml -Path .\PSKoans\Koans\dbatools\Mocks\Database_System.xml
+    } -ParameterFilter { $_.SqlInstance -eq 'localhost' -and $_.ExcludeUser -eq $true }
+
+    Mock -CommandName Get-DbaDatabase -MockWith {
+        Import-Clixml -Path .\PSKoans\Koans\dbatools\Mocks\Database_TestDb.xml
+    } -ParameterFilter { $_.SqlInstance -eq 'localhost' -and $_.ExludeSystem -eq $true }
+    #endregion
+
+    # `Get-DbaDatabase` requires one thing; A SQL Server instance name.
+    # You can pass in "localhost" for the default name for a SQL Server instance.
+    # The simplest usage of `Get-DbaDatabase` is to run it and passing in the name
+    # of the SQL Server instance. This will get information about all the databases
+    # on the instance.
+    $AllDatabases = Get-DbaDatabase -SqlInstance ____
+    $AllDatabases.Count | Should -Be 5
+
+    # By passing in the SQL Server instance and the name of a specific database, by
+    # using the `-Database` parameter, we can get information on that single
+    # database instead.
+    $MasterDatabase = Get-DbaDatabase -SqlInstance localhost -Database ____
+    $MasterDatabase.Name | Should -Be 'testdb'
+
+    # You may want to get only the system databases on an instance. While it is
+    # different databases on the instance.
+    $UserDbsExcluded = Get-DbaDatabase -SqlInstance localhost -____
+    $UserDbsExcluded | Select-Object -ExpandProperty Name | Should -BeIn 'tempdb', 'master', 'model', 'msdb'
+
+    # The same can be done to exclude system databases by providing the -ExcludeSystem
+    # parameter switch.
+    $SystemDbsExluded = Get-DbaDatabase -SqlInstance localhost -____
```

Comment:
I'm not sure this is the best way to go with these. These will throw a parameter binding exception, which will render rather oddly in the meditation prompt. Given that these are Mocks, they may even look like they come from within Pester rather than within the koan.

Reply:
This one is still pending...

Do we assume that the people know about splatting? I could create a splat and get them to do __ on the `ExcludeSystem = $true` bit? E.g.

    $Param = @{
        ExcludeSystem = ____
    }
    Get-DbaDatabase @Param

They'd have to know about switches then as well though 🤔 

Otherwise, I could either give them an options and they choose one the right one?
     
    Get-DbaDatabase -ExcludeUser
    Get-DbaDatabase -ExcludeSystem

Or else write it correctly and they read it...

It's a tough one.

Reply:
I vote splatting - it's OK to assume they know some of the basics from the rest of PSKoans here, I think. There are existing koans for that, so it should be OK

Reply:
Splatting it is!